sql/chunk.sql

```diff
@@ -27,23 +27,57 @@ $BODY$
 DECLARE
     partition_epoch_row _timescaledb_catalog.partition_epoch;
     chunk_row           _timescaledb_catalog.chunk;
-    time_interval BIGINT;
+    time_interval       BIGINT;
+    start_distinct_ctn  BIGINT;
+    end_distinct_ctn    BIGINT;
 BEGIN
+
     SELECT pe.*
     INTO partition_epoch_row
     FROM _timescaledb_catalog.partition p
     INNER JOIN _timescaledb_catalog.partition_epoch pe ON (p.epoch_id = pe.id)
     WHERE p.id = partition_id
     FOR SHARE;
 
+    -- Check if there are chunks in other partitions.
+    -- If so, use their time range to keep chunks time aligned
+
+    SELECT COUNT(DISTINCT(start_time)), COUNT(DISTINCT(end_time))
```

Comment:
There might be more than one chunk as we have to check all other partitions so that is why the distinct is there. The count and check is to verify that all matching chunks share the same constraints. I am not sure how to solve that with an INTO select ?